service/history/workflow/workflow_task_state_machine.go

```diff
@@ -1087,11 +1080,64 @@ func (m *workflowTaskStateMachine) afterAddWorkflowTaskCompletedEvent(
 	attrs := event.GetWorkflowTaskCompletedEventAttributes()
 	m.ms.executionInfo.LastCompletedWorkflowTaskStartedEventId = attrs.GetStartedEventId()
 	m.ms.executionInfo.MostRecentWorkerVersionStamp = attrs.GetWorkerVersion()
-	if err := m.ms.CompleteDeploymentTransition(attrs.GetVersioningBehavior()); err != nil {
-		return err
+
+	wftDeployment := attrs.GetDeployment()
+	wftBehavior := attrs.GetVersioningBehavior()
+	versioningInfo := m.ms.GetExecutionInfo().GetVersioningInfo()
```

Comment:
I think you can add early exit here. If `versioningInfo` is `nil`, all the rest doesn't need to be checked at all.

Reply:
ok, I read a comment about speculative WFT :-). Now I believe that you just need to move that block with `versionInfo` initialization here.

Reply:
can you make a suggested change? I'm not totally following.

Reply:
Please add a comment saying that `versionInfo` can be `nil`.